core/templates/dev/head/pages/collection_editor/editor_tab/CollectionNodeCreatorDirective.js

```diff
@@ -25,19 +25,47 @@ oppia.directive('collectionNodeCreator', [function() {
       'validatorsService', 'CollectionEditorStateService',
       'CollectionLinearizerService', 'CollectionUpdateService',
       'CollectionNodeObjectFactory', 'ExplorationSummaryBackendApiService',
-      'siteAnalyticsService',
+      'siteAnalyticsService', 'UrlInterpolationService',
       function(
           $scope, $http, $window, $filter, alertsService,
           validatorsService, CollectionEditorStateService,
           CollectionLinearizerService, CollectionUpdateService,
           CollectionNodeObjectFactory, ExplorationSummaryBackendApiService,
-          siteAnalyticsService) {
+          siteAnalyticsService, UrlInterpolationService) {
         $scope.collection = CollectionEditorStateService.getCollection();
         $scope.newExplorationId = '';
         $scope.newExplorationTitle = '';
         var CREATE_NEW_EXPLORATION_URL_TEMPLATE = '/create/<exploration_id>';
+        var SEARCH_EXPLORATION_URL_TEMPLATE = (
+          '/exploration/metadata_search?q=<search_query>');
+
+        $scope.getExplorations = function(searchQuery) {
+          if (/^[a-zA-Z0-9- ]*$/.test(searchQuery) == true) {
+            queryUrl = UrlInterpolationService.interpolateUrl(
+              SEARCH_EXPLORATION_URL_TEMPLATE, {
+                search_query: searchQuery
+              }
+            );
+            return $http.get(queryUrl).then(function(response) {
+              return response.data.collection_node_metadata_list.map(
+                function(item) {
+                return '(#' + item.id + ') ' + item.title;
+              });
+            });
+          } else {
+            alertsService.addWarning(
+              'Cannot search explorations with special characters in title.');
+          }
+        };
 
         var addExplorationToCollection = function(newExplorationId) {
+          if (newExplorationId.length > 1 && newExplorationId[0] == '(') {
```

Comment:
This is muddy. When you pass in an exploration id, it should be an exploration id (and not an "exploration id with parentheses"). Please can you refactor so that this is the case?